### PR TITLE
Task data layer

### PR DESCRIPTION
**Pull request by absmith22**
The render is still fairly empty without another component to display the tasks.

### CODE REVIEW

**Comment by absmith22** [comment]
Failing lint due to a server-side file (tasks.route.ts) that does not use a number of its imports

File: client/src/components/TaskLayer.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          tasks: temp,
+        });
+      });
+  }
+
+  /**
+   * send a POST request to the server to create a new task in tasks table, and a new
+   * entry in the approprate task table such as time_off_requests (issue #65)
+   */
+  async create_task(data: any) {
```
Comment (benmelz) [inline client/src/components/TaskLayer.tsx:49]:
```suggestion
  async createTask(data: any) {
```

File: client/src/components/TaskLayer.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      console.error(e);
+      throw new Error("Unexpected error involving POST endpoint.");
+    }
+  }
+
+  /**
+   * send a PUT request to update an existing task in tasks table, and update the
+   * subsequent task table such as time_off_requests (issue #66)
+   */
+  async update_task(data: any) {
```
Comment (benmelz) [inline client/src/components/TaskLayer.tsx:86]:
```suggestion
  async updateTask(data: any) {
```

File: client/src/components/TaskLayer.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      });
+  }
+
+  /**
+   * send a POST request to the server to create a new task in tasks table, and a new
+   * entry in the approprate task table such as time_off_requests (issue #65)
+   */
+  async create_task(data: any) {
+    try {
+      const res = await fetch("/api/sessions", {
```
Comment (benmelz) [inline client/src/components/TaskLayer.tsx:51]:
```suggestion
      const res = await fetch("/api/tasks", {
```

File: client/src/components/TaskLayer.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    this.update_task = this.update_task.bind(this);
+  }
+
+  /**
+   * send a GET request to the server to return all tasks associated with a certain
+   * user ID, including all supplemental information associated with tasks (issue #64)
+   */
+  componentDidMount() {
+    const userId = this.props.userId;
+    return fetch("/api/users", {
```
Comment (benmelz) [inline client/src/components/TaskLayer.tsx:26]:
```suggestion
    return fetch(`/api/users/${this.props.userId}/tasks`, {
```

File: client/src/components/TaskLayer.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+      method: "GET",
+      mode: "cors",
+      body: JSON.stringify({ userId }),
```
Comment (benmelz) [inline client/src/components/TaskLayer.tsx:29]:
```suggestion
```

File: client/src/components/TaskLayer.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    }
+  }
+
+  /**
+   * send a PUT request to update an existing task in tasks table, and update the
+   * subsequent task table such as time_off_requests (issue #66)
+   */
+  async update_task(data: any) {
+    try {
+      const res = await fetch("api/tasks", {
```
Comment (benmelz) [inline client/src/components/TaskLayer.tsx:88]:
```suggestion
      const res = await fetch(`/api/tasks/${data.id}`, {
```

File: client/src/components/TaskLayer.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+      console.error(e);
+      throw new Error("Unexpected error involving PUT endpoint.");
+    }
+  }
+
+  render() {
+    return (
+      <TaskContext.Provider
+        value={{
+          create_task: this.create_task,
```
Comment (benmelz) [inline client/src/components/TaskLayer.tsx:123]:
```suggestion
          tasks: this.state.tasks,
          create_task: this.create_task,
```

**Review by benmelz — CHANGES_REQUESTED** [review]

**Pull request merged**